### PR TITLE
feat(provider/kubernetes): support podSpec struct within server g…

### PR DESCRIPTION
**Pull request by chlung**
Support the new podSpec strut (#1641) for server group operations, create/run job/clone.

### CODE REVIEW

**Comment by chlung** [comment]
@lwander Can you review and provide any feedback?  No major code restructure changes which try to avoid introduce bugs.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy [hunk 1]
```diff
@@ -895,38 +930,74 @@ class KubernetesApiConverter {
...
-    if (description.restartPolicy) {
-      podTemplateSpecBuilder.withRestartPolicy(description.restartPolicy)
-    }
+    KubernetesPodSpecDescription podSpec = new KubernetesPodSpecDescription()
+    podSpec = description.podSpec
 
-    if (description.terminationGracePeriodSeconds) {
-      podTemplateSpecBuilder.withTerminationGracePeriodSeconds(description.terminationGracePeriodSeconds)
-    }
+    if ( podSpec ) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: This won't work until you migrate the UI to support podSpec in favor of/alongside the current deploy description.

Comment (chlung) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: There is another UI developer is working on UI changes.  The code should be running fine for now with original UI code base.

Comment (chlung) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: I has tested my changes with direct curl command with the podSpec structure.

Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: Ah I see - the old path is below - sounds good. 

Given that we're picking a release candidate for 1.0 today/tomorrow I need to hold off on merging this out of fear that we'll destabilize the code right when we need our release pipeline to work.

Comment (chlung) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: No problem.  Just let me know if you find out something that you would like me to make any changes.  Thanks.

Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: Thanks @chlung

Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: Could you drop the spaces around the parens on the above line?

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/job/CloneKubernetesJobAtomicOperation.groovy [hunk 2]
```diff
@@ -79,9 +81,15 @@ class CloneKubernetesJobAtomicOperation implements AtomicOperation<DeploymentRes
     newDescription.stack = description.stack ?: ancestorNames.stack
     newDescription.freeFormDetails = description.freeFormDetails ?: ancestorNames.detail
     newDescription.namespace = description.namespace ?: description.source.namespace
+
     if (!description.container) {
       newDescription.container = KubernetesApiConverter.fromContainer(ancestorPod.spec?.containers?.get(0))
     }
+    if ( KubernetesUtil.hasPodSpec(description) ) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/job/CloneKubernetesJobAtomicOperation.groovy]: Extra spaces here too

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/model/KubernetesServerGroup.groovy [hunk 3]
```diff
@@ -174,7 +181,14 @@ class KubernetesServerGroup implements ServerGroup, Serializable {
     return new ServerGroup.ImagesSummary() {
       @Override
       List<? extends ServerGroup.ImageSummary> getSummaries () {
-        deployDescription.containers.collect({ KubernetesContainerDescription it ->
+        def containers = new ArrayList<KubernetesContainerDescription>()
+        if ( KubernetesUtil.hasPodSpec(deployDescription) ) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/model/KubernetesServerGroup.groovy]: Extra spaces

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/model/KubernetesServerGroup.groovy [hunk 4]
```diff
@@ -63,8 +63,15 @@ class KubernetesServerGroup implements ServerGroup, Serializable {
   Map<String, Object> getBuildInfo() {
     def imageList = []
     def buildInfo = [:]
-    for (def container : this.deployDescription.containers) {
-      imageList.add(KubernetesUtil.getImageIdWithoutRegistry(container.imageDescription))
+
+    if ( KubernetesUtil.hasPodSpec(this.deployDescription) ) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/model/KubernetesServerGroup.groovy]: Extra spaces

**Comment by chlung** [comment]
@lwander updated per your feedback.  Thanks.

**Comment by lwander** [comment]
I think we can safely close this, let me know if we need to reopen

**Pull request closed without merging**